k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -833,3 +841,21 @@ pub(crate) async fn get_source_version(ns: &str) -> error::Result<String> {
         .to_string();
     Ok(value.to_string())
 }
+
+/// Parse set-file and append set-args
+pub(crate) async fn parse_file_values(upgrade_args: &UpgradeArgs) -> error::Result<Vec<String>> {
+    let mut args: Vec<String> = Vec::new();
+    for file in &upgrade_args.set_file {
+        let data: Vec<_> = file.split('=').collect();
```

Comment:
Its because , you upgrade-job will run on a different node from where the file is present. Upgrade job will fail to find the file on that node. So reading the file and then passing it to the job.